src/lib/validate-request.ts

```diff
@@ -23,15 +23,15 @@ export class PeprValidateRequest<T extends KubernetesObject> {
    * Provides access to the old resource in the request if available.
    * @returns The old Kubernetes resource object or null if not available.
    */
-  get OldResource() {
+  get OldResource(): T | undefined {
     return this.#input.oldObject;
   }
 
   /**
    * Provides access to the request object.
    * @returns The request object containing the Kubernetes resource.
    */
-  get Request() {
+  get Request(): AdmissionRequest<T> {
```

Comment:
Can we narrow down the value of `T` from a runtime generic to something more specific?